File: app/src/main/java/com/aavens/mindloft/db/ThingsDatabaseHelper.java
package com.aavens.mindloft.db;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class ThingsDatabaseHelper extends SQLiteOpenHelper {
    private static final int DATABASE_VERSION = 1;
    private static final String DATABASE_NAME = "Things.db";

    private static final String TEXT_TYPE = " TEXT";
    private static final String COMMA = ",";
    private static final String SQL_CREATE = "CREATE TABLE " + ThingContract.ThingEntry.TABLE_NAME + "( " +
            ThingContract.ThingEntry._ID + " INTEGER PRIMARY KEY, " +
            ThingContract.ThingEntry.COLUMN_NAME_ID + TEXT_TYPE + COMMA +
            ThingContract.ThingEntry.COLUMN_NAME_TITLE + TEXT_TYPE + COMMA +
            ThingContract.ThingEntry.COLUMN_NAME_VALUE_TYPE + TEXT_TYPE + COMMA +
            ThingContract.ThingEntry.COLUMN_NAME_VALUE + TEXT_TYPE + COMMA +
            ThingContract.ThingEntry.COLUMN_NAME_ROOM_ID + TEXT_TYPE + COMMA +
            ThingContract.ThingEntry.COLUMN_NAME_DATE + " DATETIME DEFAULT CURRENT_TIMESTAMP" +
            ")";
    private static final String SQL_DELETE = "DROP TABLE IF EXISTS " + ThingContract.ThingEntry.TABLE_NAME;

    public ThingsDatabaseHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(SQL_CREATE);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL(SQL_DELETE);
        onCreate(db);
    }

    @Override
    public void onDowngrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        onUpgrade(db, oldVersion, newVersion);
    }
}

File: app/src/main/java/com/aavens/mindloft/db/ThingsDatabaseHelper.kt
package com.aavens.mindloft.db

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class ThingsDatabaseHelper(context: Context) : SQLiteOpenHelper(context, ThingsDatabaseHelper.DATABASE_NAME, null, ThingsDatabaseHelper.DATABASE_VERSION) {

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(SQL_CREATE)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL(SQL_DELETE)
        onCreate(db)
    }

    override fun onDowngrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        onUpgrade(db, oldVersion, newVersion)
    }

    companion object {
        private val DATABASE_VERSION = 3
        private val DATABASE_NAME = "Rooms.db"

        private val TEXT_TYPE = " TEXT"
        private val COMMA = ","
        private val SQL_CREATE = "CREATE TABLE " + ThingContract.ThingEntry.TABLE_NAME + "( " +
                ThingContract.ThingEntry._ID + " INTEGER PRIMARY KEY, " +
                ThingContract.ThingEntry.COLUMN_NAME_VALUE_TYPE + TEXT_TYPE + COMMA +
                ThingContract.ThingEntry.COLUMN_NAME_VALUE + TEXT_TYPE + COMMA +
                ThingContract.ThingEntry.COLUMN_NAME_ROOM_ID + TEXT_TYPE + COMMA +
                ThingContract.ThingEntry.COLUMN_NAME_DATE + " DATETIME DEFAULT CURRENT_TIMESTAMP" +
                ")"
        private val SQL_DELETE = "DROP TABLE IF EXISTS " + ThingContract.ThingEntry.TABLE_NAME
    }
}
